gossip/monitor/monitor.go

```diff
@@ -55,84 +81,71 @@ func DefaultConfig() *Config {
 }
 
 type Monitor struct {
-	client *client.HTTPClient
-	conf   Config
-
-	metricsServer      *http.Server
-	prometheusRegistry *prometheus.Registry
-
+	client          *client.HTTPClient
+	conf            *Config
 	taskCh          chan Task
 	quitCh          chan bool
 	executionTicker *time.Ticker
 }
 
-func NewMonitor(conf Config) (*Monitor, error) {
-	// Metrics
-	metrics.QedMonitorInstancesCount.Inc()
+type Task interface {
+	Do()
+}
 
+func NewMonitor(conf *Config) (*Monitor, error) {
+	QedMonitorInstancesCount.Inc()
 	// QED client
 	transport := http.DefaultTransport.(*http.Transport)
 	transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: false}
 	httpClient := http.DefaultClient
 	httpClient.Transport = transport
-	client, err := client.NewHTTPClient(
+	qed, err := client.NewHTTPClient(
 		client.SetHttpClient(httpClient),
 		client.SetURLs(conf.QEDUrls[0], conf.QEDUrls[1:]...),
 		client.SetAPIKey(conf.APIKey),
 	)
 	if err != nil {
 		return nil, errors.Wrap(err, "Cannot start http client: ")
 	}
-
 	monitor := Monitor{
-		client: client,
+		client: qed,
 		conf:   conf,
 		taskCh: make(chan Task, 100),
 		quitCh: make(chan bool),
 	}
 
-	r := prometheus.NewRegistry()
-	metrics.Register(r)
-	monitor.prometheusRegistry = r
-	metricsMux := metricshttp.NewMetricsHTTP(r)
-
-	addr := strings.Split(conf.MetricsAddr, ":")
-	monitor.metricsServer = &http.Server{
-		Addr:    ":1" + addr[1],
-		Handler: metricsMux,
-	}
-
-	go func() {
-		log.Debugf("	* Starting metrics HTTP server in addr: %s", conf.MetricsAddr)
-		if err := monitor.metricsServer.ListenAndServe(); err != http.ErrServerClosed {
-			log.Errorf("Can't start metrics HTTP server: %s", err)
-		}
-	}()
-
 	monitor.executionTicker = time.NewTicker(conf.TaskExecutionInterval)
 	go monitor.runTaskDispatcher()
 
 	return &monitor, nil
 }
 
-type Task interface {
-	Do()
+func (m Monitor) RegisterMetrics(r *prometheus.Registry) {
```

Comment:
If you're going to change the way we register metrics IMHO, you should change every occurrence of this code, or at least make an Issue, tech debt post-it or whatever to end this task.

Reply:
It is changed in all processors.